app/src/main/java/com/x8bit/bitwarden/ui/platform/feature/settings/accountsecurity/PinInputDialog.kt

```diff
@@ -107,7 +107,11 @@ fun PinInputDialog(
                     label = stringResource(id = R.string.pin),
                     value = pin,
                     autoFocus = true,
-                    onValueChange = { pin = it },
+                    onValueChange = { newValue ->
+                        if (newValue.all { it.isDigit() }) {
+                            pin = newValue
```

Comment:
What do you think of this:

```kotlin
onValueChange = { pin = it.filter { it.isDigit() } },
```

Reply:
I guess the pros: are 86 one level of `{}`.
cons: we are technically reassigning pin each time, which wouldn't really cause recomps as the state's value wouldn't change.
either way we are iterating through whole string, so for the sack of pseudo-efficiency and laziness I would feel fine to leave it, but if the nesting is something we want to avoid this change would also be fine with me.
Your call?